### ISSUE TITLE
A small memory logger issue

### ISSUE THREAD

**Issue by YaqiWang** [idaholab/moose#8392]
### Description of the enhancement or error report
When I run memory logger:
```
moose/test/tests/kernels/2d_diffusion]> ../../../../scripts/memory_logger.py --run "../../../moose_test-opt -i 2d_diffusion_test.i"
```
and then
```
moose/test/tests/kernels/2d_diffusion]> ../../../../scripts/memory_logger.py --plot /Users/wangy2/projects/moose/test/tests/kernels/2d_diffusion/2d_diffusion_test_memory.log
```
I got 
```
/opt/moose/miniconda/lib/python2.7/site-packages/matplotlib/font_manager.py:273: UserWarning: Matplotlib is building the font cache using fc-list. This may take a moment.
  warnings.warn('Matplotlib is building the font cache using fc-list. This may take a moment.')
Traceback (most recent call last):
  File "../../../../scripts/memory_logger.py", line 1176, in <module>
    MemoryPlotter(args)
  File "../../../../scripts/memory_logger.py", line 740, in __init__
    self.buildGraph()
  File "../../../../scripts/memory_logger.py", line 816, in buildGraph
    tmp_memory.append(decimal.Decimal(records[1]) / self.multiples)
  File "/opt/moose/miniconda/lib/python2.7/decimal.py", line 1321, in __truediv__
    return context._raise_error(DivisionUndefined, '0 / 0')
  File "/opt/moose/miniconda/lib/python2.7/decimal.py", line 3873, in _raise_error
    raise error(explanation)
decimal.InvalidOperation: 0 / 0
```

### Rationale for the enhancement or information for reproducing the error
I am guessing this problem is too small and is causing this issue.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
None.

**Comment by permcody** [comment]
I think the problem here is that if a program runs too fast, there will only be a single sample. There's some problem with the logic in handling that case. 

To replicate, try deleting all the entries in the log file but the first.

Tag @milljm

**Comment by milljm** [comment]
@permcody 
Thank you for taking a look... Now that you mention it, yes I most certainly have to wait a few moments (seconds) before attempting to take the first sample, because on an job scheduling system, I realized nothing starts as soon as you tell it to. The memory_logger will accidentally assume the process has exited if I try to take the first sample to quickly.

**Comment by permcody** [comment]
We just need to not divide by zero in the code. It's a minor issue.

**Comment by permcody** [comment]
Fixed by #8526
